admin/admin.go

```diff
@@ -94,3 +94,14 @@ func (s *Service) Close() error {
 
 	return s.DB.Close()
 }
+
+// NewMock for testing
+func NewMock(db database.DB, githubClient *github.Client, prov provisioner.Provisioner, issuer *auth.Issuer, emailClient *email.Client) *Service {
+	return &Service{
+		DB:          db,
+		github:      githubClient,
+		provisioner: prov,
+		issuer:      issuer,
+		Email:       emailClient,
+	}
+}
```

Comment:
It would be better if we could just use `New` and pass in mock dependencies where applicable. It would ensure better consistency, especially as we increase the functionality in this service in the future.

The only thing preventing that appears to be the Github client. However, it's used very little, and we use a very small part of its functionality. What about creating a `type Github interface` (in `admin/github.go`) with just the two or three functions we need, having a real and a mock implementation of it, and then passing `github Github` as a parameter to `admin.New` instead?

Reply:
tried to abstract out Github using interface
```
type Github interface {
	GetGithubInstallation(ctx context.Context, githubURL string) (int64, error)
	LookupGithubRepoForUser(ctx context.Context, installationID int64, githubURL, gitUsername string) (*github.Repository, error)
	ProcessGithubEvent(ctx context.Context, rawEvent any) error
}
```
Implementation of this creates the github client using github app id and key. However this calls needs DB as well. DB can be create before admin.Service creation and can be passed to both service and Github implementation. However, to process github push it needs to reconcile for which it needs `admin.Service` thus creating cyclic dependency. Any thoughts here ?

Reply:
I would make the `Github` interface even simpler, just pure proxy to Github API. So anything involving our DB stays in the `admin.Service`. Is that not possible?

Reply:
change, please check